virttest/qemu_devices/qcontainer.py

```diff
@@ -3293,11 +3293,7 @@ def thread_context_define_by_params(self, params, name):
         Create thread-context object from params.
         """
         tc_params = Params()
-        prefix = "vm_thread_context_"
-        for key in params:
-            if key.startswith(prefix):
-                new_key = key.rsplit(prefix)[1]
-                tc_params[new_key] = params[key]
+        tc_params["node-affinity"] = params["vm_thread_context_node-affinity_%s" % name]
```

Comment:
```suggestion
        tc_params["node-affinity"] = params["vm_thread_context_node-affinity"]
```

since here the `vm_thread_context_node-affinity` is already a per-object param.

Reply:
Correct, code updated !